Error: Library not loaded: @rpath/libunwind.1.dylib on macOS M2

Description:
I encountered an error while running the project on my MacBook Pro M2 (Apple Silicon). The error message is as follows:

```bash
dyld[70274]: Library not loaded: @rpath/libunwind.1.dylib
Referenced from: <BE8B7E7B-8446-3C43-BA2E-B6EAA7E8D61F> /private/var/tmp/_bazel_*****/bebcbb61406245ffe61733b8c2f2ccde/execroot/sondb/bazel-out/darwin_arm64-dbg/bin/src/sondb
Reason: tried: '/usr/local/lib/libunwind.1.dylib' (no such file), '/usr/lib/libunwind.1.dylib' (no such file, not in dyld cache)
make: *** [debug-run] Error 255
```
It seems to be related to a missing or incorrect library (libunwind) on macOS M2. I believe this issue is specific to the project configuration.

 
Environment:

macOS version:13.5.2 ventura (m2 max)
Project version:0.10.3
LLVM: 15.0.6/15.0.7/16.0.0
 

I would appreciate any help in resolving this issue. Thank you!

--- a/toolchain/cc_toolchain_config.bzl
+++ b/toolchain/cc_toolchain_config.bzl
@@ -270,19 +270,30 @@ def cc_toolchain_config(
             # libc++abi, so static linking them becomes a problem. We need to
             # ensure that they are dynamic linked from the system sysroot and
             # not static linked from the toolchain, so explicitly have the
-            # sysroot directory on the search path and then add the toolchain
-            # directory back after we are done.
+            # sysroot directory on the search path.
+            #
+            # The toolchain lib directory is intentionally NOT added to the
+            # search path here. In sandboxed execution, the toolchain's lib/
+            # directory is empty (only declared outputs are present), so the
+            # previous -L flag was a harmless no-op. However, with
+            # --spawn_strategy=local, the full toolchain lib/ directory is
+            # visible to the linker, and ld64 discovers dylibs like
+            # libunwind.1.dylib via the -L search path. These get baked into
+            # the binary as LC_LOAD_DYLIB entries with @rpath install names
+            # that fail at runtime because the toolchain directory is not in
+            # the binary's @rpath search path.
+            #
+            # libunwind_link_flags is left empty on macOS because libunwind
+            # is unconditionally provided by libSystem.B.dylib (clang always
+            # passes -lSystem via Darwin.cpp). The toolchain's libunwind is
+            # redundant and its dylib causes the runtime failure described
+            # above, so the libunwind config flag has no effect on macOS.
             link_flags.extend([
                 "-L{}/usr/lib".format(sysroot_path),
                 "-lc++",
                 "-lc++abi",
                 "-Bdynamic",
-                "-L{}lib".format(toolchain_path_prefix),
             ])
-            libunwind_link_flags = [
-                "-Bstatic",
-                "-lunwind",
-            ]
         else:
             # For single-platform builds, we can statically link the bundled
             # libraries.

